sdcm/cluster.py

```diff
@@ -1575,173 +1575,99 @@ def config_setup(self, seed_address=None, cluster_name=None, enable_exp=True, en
-            pattern = re.compile('murmur3_partitioner_ignore_msb_bits:.*')
-            if pattern.findall(scylla_yaml_contents):
-                scylla_yaml_contents = pattern.sub('murmur3_partitioner_ignore_msb_bits: {0}'.format(murmur3_partitioner_ignore_msb_bits),
-                                                   scylla_yaml_contents)
-            else:
-                scylla_yaml_contents += "\nmurmur3_partitioner_ignore_msb_bits: {0}\n".format(
-                    murmur3_partitioner_ignore_msb_bits)
+            scylla_yml['murmur3_partitioner_ignore_msb_bits'] = int(murmur3_partitioner_ignore_msb_bits)
 
         if enable_exp:
-            pattern = re.compile('experimental:.*')
-            if pattern.findall(scylla_yaml_contents):
-                scylla_yaml_contents = pattern.sub('experimental: true', scylla_yaml_contents)
-            else:
-                scylla_yaml_contents += "\nexperimental: true\n"
+            scylla_yml['experimental'] = True
 
         if endpoint_snitch:
-            pattern = re.compile('endpoint_snitch:.*')
-            scylla_yaml_contents = pattern.sub('endpoint_snitch: "{0}"'.format(endpoint_snitch),
-                                               scylla_yaml_contents)
+            scylla_yml['endpoint_snitch'] = endpoint_snitch
+
         if not client_encrypt:
-            pattern = re.compile('.*enabled: true.*# <client_encrypt>.*')
-            scylla_yaml_contents = pattern.sub(
-                '   enabled: false                    # <client_encrypt>', scylla_yaml_contents)
+            scylla_yml['client_encryption_options'] = dict(enabled=False)
 
         if self.enable_auto_bootstrap:
-            if 'auto_bootstrap' in scylla_yaml_contents:
-                if re.findall("auto_bootstrap: False", scylla_yaml_contents):
-                    self.log.debug('auto_bootstrap is not set as expected, update it to `True`.')
-                pattern = re.compile('auto_bootstrap:.*')
-                scylla_yaml_contents = pattern.sub('auto_bootstrap: True',
-                                                   scylla_yaml_contents)
-            else:
-                self.log.debug('auto_bootstrap is missing, set it `True`.')
-                scylla_yaml_contents += "\nauto_bootstrap: True\n"
+            scylla_yml['auto_bootstrap'] = True
         else:
-            if 'auto_bootstrap' in scylla_yaml_contents:
-                if re.findall("auto_bootstrap: True", scylla_yaml_contents):
-                    self.log.debug('auto_bootstrap is not set as expected, update it to `False`.')
-                pattern = re.compile('auto_bootstrap:.*')
-                scylla_yaml_contents = pattern.sub('auto_bootstrap: False',
-                                                   scylla_yaml_contents)
-            else:
-                self.log.debug('auto_bootstrap is missing, set it `False`.')
-                scylla_yaml_contents += "\nauto_bootstrap: False\n"
```

Comment:
@fruch how from all this logic you decided that this if,else is enough?
Apparently, it's not.

This is what I see in the log of one cluster with an issue:
```
Node longevity-lwt-3h-4-0-db-node-5cfd2305-1 [18.207.221.117 | 172.30.0.162] (seed: False): Scylla YAML configuration:
api_address: 127.0.0.1
api_doc_dir: /opt/scylladb/api/api-doc/
api_port: 10000
api_ui_dir: /opt/scylladb/swagger-ui/dist/
auto_bootstrap: false
batch_size_fail_threshold_in_kb: 50
batch_size_warn_threshold_in_kb: 5
broadcast_rpc_address: 172.30.0.162
cas_contention_timeout_in_ms: 1000
client_encryption_options:
  enabled: false
cluster_name: longevity-lwt-3h-4-0-db-cluster-5cfd2305
commitlog_segment_size_in_mb: 32
commitlog_sync: periodic
commitlog_sync_period_in_ms: 10000
commitlog_total_space_in_mb: -1
endpoint_snitch: org.apache.cassandra.locator.Ec2Snitch
experimental: true
listen_address: 172.30.0.162
murmur3_partitioner_ignore_msb_bits: 12
native_transport_port: 9042
num_tokens: 256
partitioner: org.apache.cassandra.dht.Murmur3Partitioner
read_request_timeout_in_ms: 5000
rpc_address: 172.30.0.162
rpc_port: 9160
seed_provider:
- class_name: org.apache.cassandra.locator.SimpleSeedProvider
  parameters:
  - seeds: 172.30.0.162
write_request_timeout_in_ms: 2000

```
If you see here, somehow the auto_bootstrap is set to "false" (lower case) when the cluster starts.
(We don't need to set anything at all, but we should start with the default scylla.yaml which now doesn't even have this parameter by default).

Then, in many place we are looking to "sed" False with True, but probably it doesn't do anything now.

I don't know where the lowercase "false" is coming from, but somehow changing the above logic caused issue.
we need to fix it ASAP.

@bentsi FYI

Reply:
created an urgent task task: https://trello.com/c/Djp6Oht5/1844-fix-autobootstrap-logic

Reply:
YAML true, is always lower case.

where is the code that sed for "False", it should be replaced, reading correctly the yaml... or using this function

Reply:
thanks, it's related to scylla issue https://github.com/scylladb/scylla/issues/6120 and https://github.com/scylladb/scylla/issues/6121

Reply:
this should fix it: 
https://github.com/scylladb/scylla-cluster-tests/pull/2031